### PR TITLE
DAS-2034 - Remove custom retry logic from HOSS in favour of harmony-service-lib.

### PR DESCRIPTION
**Pull request by owenlittlejohns**
## Description

This ticket removes custom retry logic for downloading a file from OPeNDAP. The `harmony-service-lib` also has retry logic, as does each step in a Harmony workflow. All three of these retry types currently can lead to 150 attempts to download a file, which is currently causing issues for OPeNDAP with the GHRSST MUR data.

## Jira Issue ID

DAS-2034.

## Local Test Steps

* Build this branch locally using:
```
./bin/build-image
```
* Restart your local Harmony instance:
```
# If Harmony is already running:
kubectl delete ns harmony

# In the Harmony repository:
./bin/bootstrap-harmony
```
* Run through the tests in the [HOSS regression test notebook](https://github.com/nasa/harmony-regression-tests/tree/main/test/hoss) (for local testing, you'll need to run the notebook in your browser, not via the `run_notebooks.sh` script). This will also mean setting `harmony_host_url = 'http://localhost:3000'` in the notebook. The tests should pass.
* Bonus check - the logs for shape file subsets contain logging indicating that a shape file has been downloaded. The logging can be accessed via the `http://localhost:3000/workflow-ui` endpoint, and the test to check uses the collection concept ID 'C1245618475-EEDTEST' and has three steps in the workflow: query-cmr, harmony-opendap-subsetter and maskfill. The loggin will contain a line similar to:
```
"2023-12-20 03:51:10,810 [INFO] [harmony-service.get_request_shape_file:61] Downloading request shape file\r\n",
```

## PR Acceptance Checklist
* [x] Jira ticket acceptance criteria met.
* [x] `CHANGELOG.md` updated to include high level summary of PR changes.
* [x] `VERSION` updated if publishing a release.
* [x] Tests added/updated and passing.
* ~~Documentation updated (if needed).~~

### CODE REVIEW

File: hoss/bbox_utilities.py [hunk 1]
```diff
@@ -58,6 +58,7 @@ def get_request_shape_file(message: Message, working_dir: str,
             raise UnsupportedShapeFileFormat(message.subset.shape.type)
 
         shape_file_url = message.subset.shape.process('href')
+        adapter_logger.info('Downloading request shape file')
```
Comment (owenlittlejohns) [inline hoss/bbox_utilities.py:61]: Initially I included the path to the shape file, but this is somewhat meaningless as Harmony creates a temporary file, which obfuscates any original user naming convention (the file name is essentially a generated ID).

**Comment by owenlittlejohns** [comment]
@lyonthefrog - I couldn't add you to the HOSS reviewers (you may need to tweak your NAMS request to include this repo). But I thought you would want to track this as someone also working on HOSS now.

@jgallagher59701 and @ndp-opendap - for your awareness. (I'm getting a little ahead of ticket process here, but wanted to start getting this change propagated out as a quick win to help reduce load on OPeNDAP)

**Review by flamingbear — APPROVED** [review]
Looks great to me.  I was able to run the regression test locally without issue. I'll approve, but there is one comment to fix in the tests before you merge.

File: tests/unit/test_utilities.py [hunk 2]
```diff
@@ -92,10 +91,18 @@ def test_download_url(self, mock_util_download):
             mock_util_download.side_effect = [self.harmony_500_error,
```
[the remark this answers is not in the document]
Comment (owenlittlejohns) [reply tests/unit/test_utilities.py:89]: Good catch. I updated that test name (and also the doc string for the overall test that mentioned retries would be tested)

**Pull request merged**